app/src/main/java/com/github/gotify/Utils.java

```diff
@@ -52,4 +69,110 @@ public static JSON json() {
     public interface DrawableReceiver {
         void loaded(Drawable drawable);
     }
+
+    public static InputStream stringToInputStream(String str) {
+        if (str == null) return null;
+        return new Buffer()
+                .writeUtf8(str)
+                .inputStream();
+    }
+
+    public static String readFileFromStream(@NonNull InputStream inputStream) {
+        StringBuilder sb = new StringBuilder();
+        String currentLine;
+
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream))) {
+            while ((currentLine = reader.readLine()) != null) {
+                sb.append(currentLine).append("\n");
+            }
+        } catch (IOException e) {
+            throw new IllegalArgumentException("failed to read input");
+        }
+
+        return sb.toString();
+    }
+
+    /////////////////////////
+    ///// SSL Utilities /////
+    /////////////////////////
+    public static Certificate parseCertificate(String cert) {
+        try {
+            CertificateFactory certificateFactory = CertificateFactory.getInstance("X509");
+
+            return certificateFactory.generateCertificate(Utils.stringToInputStream(cert));
+        } catch (Exception e) {
+            throw new IllegalArgumentException("certificate is invalid");
+        }
+    }
+
+    public static class SSLSettings {
+        boolean validateSSL;
+        String cert;
+
+        public SSLSettings(boolean validateSSL, String cert) {
+            this.validateSSL = validateSSL;
+            this.cert = cert;
+        }
+    }
+
+    // TrustManager that trusts all SSL Certs
+    private static final TrustManager trustAll = new X509TrustManager() {
+        @SuppressLint("TrustAllX509TrustManager")
+        @Override
+        public void checkClientTrusted(X509Certificate[] chain, String authType) {}
+        @SuppressLint("TrustAllX509TrustManager")
+        @Override
+        public void checkServerTrusted(X509Certificate[] chain, String authType) {}
+        @Override
+        public X509Certificate[] getAcceptedIssuers() { return new X509Certificate[]{}; }
+    };
+
+    public static void applySslSettings(OkHttpClient.Builder builder, SSLSettings settings) {
```

Comment:
Hmm that sucks, then let's keep it and I'll try to fix that later (maybe by generating an other client which uses okhttp3)

Can you change the code to this, to flow is understandable (in my opinion)?
```java
public static void applySslSettings(OkHttpClient.Builder builder, SSLSettings settings) {
    // Modified from ApiClient.applySslSettings in the client package.

    try {
        if (!settings.validateSSL) {
            SSLContext context = SSLContext.getInstance("TLS");
            context.init(
                    new KeyManager[] {}, new TrustManager[] {trustAll}, new SecureRandom());
            builder.sslSocketFactory(context.getSocketFactory(), trustAll);
            builder.hostnameVerifier((a, b) -> true);
            return;
        }

        if (settings.cert != null) {
            TrustManager[] trustManagers = certToTrustManager(settings.cert);

            if (trustManagers != null && trustManagers.length > 0) {
                SSLContext context = SSLContext.getInstance("TLS");
                context.init(new KeyManager[] {}, trustManagers, new SecureRandom());
                builder.sslSocketFactory(
                        context.getSocketFactory(), (X509TrustManager) trustManagers[0]);
                builder.hostnameVerifier((a, b) -> true);
            }
        }
    } catch (Exception e) {
        // We shouldn't have issues since the cert is verified on login.
        Log.e("Failed to apply SSL settings", e);
    }
}

private static TrustManager[] certToTrustManager(String cert) throws GeneralSecurityException {
    CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
    Collection<? extends Certificate> certificates =
            certificateFactory.generateCertificates(stringToInputStream(cert));
    if (certificates.isEmpty()) {
        throw new IllegalArgumentException("expected non-empty set of trusted certificates");
    }
    KeyStore caKeyStore = newEmptyKeyStore();
    int index = 0;
    for (Certificate certificate : certificates) {
        String certificateAlias = "ca" + Integer.toString(index++);
        caKeyStore.setCertificateEntry(certificateAlias, certificate);
    }
    TrustManagerFactory trustManagerFactory =
            TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
    trustManagerFactory.init(caKeyStore);
    return trustManagerFactory.getTrustManagers();
}
```
I changed the type of trustAll to X509TrustManager.

(untestet code)

Reply:
Added requested implementation